client/my-sites/comments/comment/comment-html-editor/add-link-dialog.jsx

```diff
@@ -0,0 +1,138 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import FormCheckbox from 'components/forms/form-checkbox';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+
+const REGEXP_EMAIL = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i;
+const REGEXP_URL = /^(https?|ftp):\/\/[A-Z0-9.-]+\.[A-Z]{2,4}[^ "]*$/i;
+const REGEXP_STANDALONE_URL = /^(?:[a-z]+:|#|\?|\.|\/)/;
```

Comment:
so it's okay to use these kinds of RegExps but we need to be explicit with their boundaries - at a MINIMUM an explanation of why we're using degraded parsers and where they will fail.

typically I like to make it obvious in the naming…

```js
const BAD_REGEXP_EMAIL = /…/i;
const BAD_REGEXP_URL = /…/i;
const BAD_STANDALONE_URL = /…/;
```

There are URL parsers we can use, including the `url` package which is in use already around the app, but those run into the problem that almost everything is a valid URL in some sense.

It may be helpful to be pretty lenient here. Did these come from somewhere else?

Also, how is a leading dot a URL? (that's the `\.` alternation value in the standalone pattern)

Reply:
> Did these come from somewhere else?

Yup: https://github.com/Automattic/wp-calypso/blob/master/client/components/tinymce/plugins/wplink/dialog.jsx#L34-L36

These regexes also appears in the post editor HTML toolbar's link dialog.
The idea was to have a behaviour 100% identical to the visual toolbar / TinyMCE link dialog.

I'm not opposed to update/fix them, but I believe we should do it in a follow up PR, tackling all 3 occurrences (MCE plugin, post editor HTML toolbar, comment editor HTML toolbar) at once.